lib/file-source/src/file_server.rs

```diff
@@ -433,6 +441,27 @@ impl Fingerprinter {
     }
 }
 
+fn fingerprinter_read_until(mut r: impl Read, delim: u8, mut buf: &mut [u8]) -> io::Result<()> {
+    while !buf.is_empty() {
```

Comment:
This looked like a bug at first read (we want to keep reading until the buffer is full, not until it's simply not empty), but I see now that it's correct since we rebind `buf` before looping. It'd be nice if we could express this more clearly, but I don't consider it a blocker.

Reply:
FYI, it's based on https://doc.rust-lang.org/src/std/io/mod.rs.html#765
I'll try to improve it though.

Reply:
Makes sense. It's probably not worth spending extra time on.